src/Filesystem/Asset.php

```diff
@@ -38,7 +38,7 @@ public function __construct(string $path)
 		$this->setProperties([
 			'path' => dirname($path),
 			'root' => $this->kirby()->root('index') . '/' . $path,
-			'url'  => $this->kirby()->url('index') . '/' . $path
+			'url'  => rtrim($this->kirby()->url('index'), '/') . '/' . $path
```

Comment:
This solution seems correct. But I think seems still confusing. `$this->kirby()->url('index')` should returns stable/consist value that with or without slash. So we shouldn't need either the `rtrim` method or the extra slash `/` 🤔

Reply:
`$this->kirby()->url('index')` by default always returns a value without trailing slash, unless it's `/`. This is handled by `Core::urls()` and `Environment::baseUrl()`.

`rtrim()` was the simplest solution I came up with (in the amount of code) and also prevents regressions in case the `index` URL for some reason ever *does* contain a trailing slash.

Open for suggestions though. :)

Reply:
I agree with you and favor the simple solution if a more general but stable solution will cause regressions 👍

Reply:
I'm good overview for this parts. But I just wonder that why didn't we do the same for `index` while the `base` url doesn't end with slash. I guess @bastianallgeier have an idea about that.
https://github.com/getkirby/kirby/blob/cfac315bc685193d0fe1b10fe3bf45a96e48fd15/src/Cms/Core.php#L459-L475

Reply:
Thinking about it, maybe we should convert `/` to an empty string already in `baseUrl`. Then it would be really consistent. But not sure if this would break other parts of the system.

Reply:
Then it seems like the safest way to go with your simple solution as this PR, do I understand right? 

But I think we need one more change: 
https://github.com/getkirby/kirby/blob/fix/4508-asset-url-double-slash/src/Panel/Document.php#L279

Reply:
To be honest, the cleanest and technically correct solution would be if all URLs always had a trailing slash. But I agree that this seems to be the most stable version to keep everything as it is.

Reply:
You are right, a trailing slash really makes the most sense.
Do you think it's just risky in general as the whole system needs to be adapted or would it be a breaking change that users and plugin devs would also need to adapt to?

In any case, the switch to a trailing slash is only something we can do in a major release. If you agree, I can verify all uses of the index URL to make sure that they support the special `/` value. Then we can already fix the bug in 3.7.4.